### PR TITLE
Improve ast::make

### PR DESCRIPTION
**Pull request by alibektas**
Add `ty_alias` and make `impl_trait` , `fn` and `impl_` have more coverage.

### CODE REVIEW

**Comment by alibektas** [comment]
At this point I am looking for a practical way to test these functions as they got rapidly more complex than I expected. This PR is simply a precursor to (#14386). So when that issue is solved I expect to uncover some mistakes as well so it is better to solve them now than later.

**Comment by Veykril** [comment]
These functions are only really tested by assists and the like and that shouldn't need to change, given we are going to code gen these functions at some point in the future anyways.

File: crates/syntax/src/ast/make.rs [hunk 1]
```diff
@@ -203,33 +196,106 @@ pub fn assoc_item_list() -> ast::AssocItemList {
...
+// FIXME : We must make *_gen_args' type ast::GenericArgList but in order to do so we must implement in `edit_in_place.rs`
+// `add_generic_arg()` just like `add_generic_param()`
+// is implemented for `ast::GenericParamList`
 pub fn impl_trait(
-    trait_: ast::Path,
-    ty: ast::Path,
-    ty_params: Option<ast::GenericParamList>,
+    is_unsafe: bool,
+    trait_gen_params: Option<ast::GenericParamList>,
+    trait_gen_args: Option<ast::GenericParamList>,
```
Comment (Veykril) [inline crates/syntax/src/ast/make.rs:252]:
```suggestion
    trait_gen_args: Option<ast::GenericArgList>,
```

File: crates/syntax/src/ast/make.rs [hunk 2]
```diff
@@ -203,33 +196,106 @@ pub fn assoc_item_list() -> ast::AssocItemList {
...
+    type_gen_params: Option<ast::GenericParamList>,
+    type_gen_args: Option<ast::GenericParamList>,
```
Comment (Veykril) [inline crates/syntax/src/ast/make.rs:254]:
```suggestion
    type_gen_args: Option<ast::GenericArgList>,
```

**Pull request merged**